packages/template-retail-react-app/app/pages/account/profile.jsx

```diff
@@ -43,56 +52,81 @@ const Skeleton = ({children, height, width, ...rest}) => {
         : {}
 
     return (
-        <ChakraSkeleton isLoaded={isRegistered} {...rest} {...size}>
+        <ChakraSkeleton isLoaded={!!customer} {...rest} {...size}>
             {children}
         </ChakraSkeleton>
     )
 }
 
 const ProfileCard = () => {
     const {formatMessage} = useIntl()
-    const customer = useCustomer()
+
+    const customerId = useCustomerId()
+    const {isRegistered} = useCustomerType()
+    const {data: customer} = useCustomer({customerId}, {enabled: !!customerId && isRegistered})
```

Comment:
In general I would have expected that there is a code reduction when using the hooks library or at a minimum no change line count as that was part of our goal, _to move some complexities out of the project and into the commerce-sdk-react_. 

From the looks of it things have gotten a little more complex, one thing that is sticking out at me is that it's not easy to the customer. You have to get the id, then get the customer object, and we are also twisting in the use of this useCustomerType hook.

The funny thing is that I'm sure all the customer data is somewhere already. Have you thought of creating a utility hook in the commerce-sdk-react? 

We previously talked about using some kind of app state. I think this would be a good reason to do that now. 

Off of the top of me head you might have a usage like this:

```
const {customer, basket, others} = useAppState()
```

Thoughts?

Reply:
@bendvc To avoid scope creep in this PR, I've created an appStateProvider and hook to handle customer info in there for now. This will help us to achieve DRY for the customer. I'll create a ticket to move basket info in there, and we will discuss whether we want to move this provider inside commerce-react-sdk

Reply:
@kevinxh and I had a chat about useAppState for many data and have some concerns about performance issue if we aim to put many data together. Let's keep the scope to a single Provider `CustomerProvider`, and we are likely to have a discussion around how we want to manage state in another ticket